### PR TITLE
Fix pylint issues and apply improvements

### PR DESCRIPTION
**Pull request by dominik003**
This PR resolves the remaining pylint errors, so we can finally include pylint in our pre-commit hook

### CODE REVIEW

File: backend/capellacollab/sessions/files/routes.py [hunk 1]
```diff
@@ -61,29 +53,29 @@ def get_files(
 
 
 @router.post("/", dependencies=[Depends(check_session_belongs_to_user)])
```
Comment (MoritzWeber0) [inline backend/capellacollab/sessions/files/routes.py:55]: `check_session_belongs_to_user` can be removed globally if `get_current_session` is used everywhere.

File: backend/capellacollab/sessions/operators/k8s.py [hunk 2]
```diff
@@ -843,29 +843,39 @@ def _get_claim_name(self, username: str) -> str:
 
     def _delete_deployment(self, name: str) -> client.V1Status:
         try:
-            return self.v1_apps.delete_namespaced_deployment(name, namespace)
+            status: client.V1Status = (
+                self.v1_apps.delete_namespaced_deployment(name, namespace)
+            )
         except exceptions.ApiException:
             log.exception("Error deleting deployment with name: %s", name)
+        return status
```
Comment (MoritzWeber0) [inline backend/capellacollab/sessions/operators/k8s.py:851]:
```
    def _delete_deployment(self, name: str) -> client.V1Status | None:
        try:
            return self.v1_apps.delete_namespaced_deployment(name, namespace)
        except exceptions.ApiException:
            log.exception("Error deleting deployment with name: %s", name)
```

Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py:851]: Also check the log lines in `kill_sessions`.

Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py:851]: And apply for all delete methods

File: backend/capellacollab/sessions/routes.py [hunk 3]
```diff
@@ -206,17 +218,21 @@ def request_session(
...
-        git_model = next(
-            gm for gm in model.git_models if gm.id == entry.git_model_id
-        )
+        if not (
+            git_model := next(
+                (gm for gm in model.git_models if gm.id == entry.git_model_id),
+                None,
+            )
+        ):
+            return
```
Comment (MoritzWeber0) [inline backend/capellacollab/sessions/routes.py:230]:
```suggestion
            continue
```

File: helm/config/backend_schema.json [hunk 4]
```diff
@@ -0,0 +1,178 @@
+{
```
Comment (MoritzWeber0) [inline helm/config/backend_schema.json:1]: Please delete this file.

File: helm/config/backend_schema.json.license [hunk 5]
```diff
@@ -0,0 +1,2 @@
+SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
```
Comment (MoritzWeber0) [inline helm/config/backend_schema.json.license:1]: Please delete this file.

File: helm/templates/backend/backend.configmap.yaml [hunk 6]
```diff
@@ -11,3 +11,5 @@ metadata:
 data:
   config.yaml: |
     {{- tpl (.Files.Get "config/backend.yaml") . | nindent 4 }}
+  config_schema.json: |
+    {{- tpl (.Files.Get "config/backend_schema.json") . | nindent 4 }}
```
Comment (MoritzWeber0) [inline helm/templates/backend/backend.configmap.yaml:15]:
```suggestion
```

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: backend/capellacollab/sessions/files/routes.py [hunk 7]
```diff
@@ -25,13 +24,10 @@
 
 
 def check_session_belongs_to_user(
-    session_id: str,
-    db: Session = Depends(get_db),
+    session: DatabaseSession = Depends(get_existing_session),
     token=Depends(JWTBearer()),
```
[the remark this answers is not in the document]
Comment (dominik003) [reply backend/capellacollab/sessions/files/routes.py:28]: Removed `check_session_belongs_to_user` and now use `get_existing_session` at all places

File: backend/capellacollab/sessions/files/routes.py (hunk 1 above)
Comment (dominik003) [reply backend/capellacollab/sessions/files/routes.py:55]: Done

File: backend/capellacollab/projects/models.py [hunk 8]
```diff
@@ -34,7 +34,7 @@ class Project(BaseModel):
     users: UserMetadata
 
     @validator("users", pre=True)
-    def transform_users(
+    def transform_users(  # pylint: disable=no-self-argument
```
[the remark this answers is not in the document]
Comment (dominik003) [reply backend/capellacollab/projects/models.py:37]: Tested it with the other added `@classmethod` and works there

**Review by MoritzWeber0 — APPROVED** [review]

**Pull request merged**